File: Javascript/calculadora.js
class Calculadora {
    constructor (Operaciones) {
        this.operaciones = Operaciones;
    }
    calcular(a, b, simbolo) {
        const operacion = this.operaciones[simbolo];
        if (operacion === undefined ) {
            return("No definido");
        }
    try {
        const resultado = operacion.ejecutar(a, b);
        return resultado;
    } catch (e) {
        return e.message 
    }
    }
}
module.exports = Calculadora;

File: C#/Calculadora.cs
using System;
using System.Collections.Generic;
public class Calculadora
{
    private Dictionary<string, IOperacion> operaciones;
    public Calculadora(Dictionary<string, IOperacion> operaciones)
    {
       this.operaciones = operaciones;
    }
    public string Calcular(double a, double b, string simbolo)
    {
        if(!operaciones.ContainsKey(simbolo))
        {
            return "Operación no válida";
        }
        IOperacion operacion = operaciones[simbolo];
        try{
            double resultado = operacion.Ejecutar(a, b);
            return resultado.ToString();
            }
            catch(DivideByZeroException e)
            {
                return e.Message;
                }
                }

}
